In [1]:
import numpy as np
from scipy import stats
from scipy.stats import binom
from scipy.stats import poisson
from scipy.stats import t
import collections
import math

# Problem Statement 1

In each of the following situations, state whether it is a correctly stated hypothesis testing problem and why?
1. H0: μ = 25, H1: μ ≠ 25
2. H0: σ > 10, H1: σ = 10
3. H0: x = 50, H1: x ≠ 50
4. H0: p = 0.1, H1: p = 0.5
5. H0: s = 30, H1: s > 30

## The three forms of Hypothesis tests are

* Form 1 - Exact vs Inexact
* Form 2 - At least vs Less than
* Form 3 - At most vs Greater than

|Form 1 - Exact vs Inexact|Form 2 - At least vs Less than|Form 3 - At most vs Greater than|
|:-:|:-:|:-:|
| $H _{0} : \mu = \mu _{0}$|$H _{0} : \mu \geq \mu _{0}$ | $H _{0} : \mu \leq \mu _{0}$|
| $H _{A} : \mu \neq  \mu _{0}$|$H _{A} : \mu < \mu _{0}$ | $H _{A} : \mu > \mu _{0}$|

### H0: μ = 25, H1: μ ≠ 25

This is a **Valid Hypothesis** of type Form 1

### H0: σ > 10, H1: σ = 10

The alternate hypothesis can never be $ = $, so this is **not a valid hypothesis**

### H0: x = 50, H1: x ≠ 50
This is a **valid Hypothesis** of type Form 1

### H0: p = 0.1, H1: p = 0.5

The alternate hypothesis can never be  = , so this is **not a valid hypothesis**

### H0: s = 30, H1: s > 30

If the Null hypothesis is $=$, the alternate hypothesis should be $\ne$, so this is **not a valid hypothesis**

The other option for a valid hypothesis is that H0: s $\leq$ 30, H1: s > 30


***
***

# Problem Statement 2
The college bookstore tells prospective students that the average cost of its textbooks is Rs. 52 with a standard deviation of Rs. 4.50. 

A group of smart statistics students thinks that the average cost is higher. 

To test the bookstore’s claim against their alternative, the students will select a random sample of size 100. 

Assume that the mean from their random sample is Rs. 52.80. 

Perform a hypothesis test at the 5% level of significance and state your decision.

### Hypothesis 
H0: μ $\leq$ 52

H1: μ $>$ 52

### Significance level
Significance Level = 5%

### Test Statics
$\mu = 52$

$\sigma = 4.50$

n = 100

$\bar {x}$ = 52.80

In [2]:
z = (52.80 - 52)/ (4.5 / 10)

print(f"P(Z > {z}) = {1 - stats.norm.cdf(z)}")

P(Z > 1.7777777777777715) = 0.0377201798134007


This is an upper tailed z test. Since 0.0377 is less than 0.05, **we reject the Null Hypothesis.**

***
***

# Problem Statement 3
A certain chemical pollutant in the Genesee River has been constant for several years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. 

A group of factory representatives whose companies discharge liquids into the river is now claiming that they have lowered the average with improved filtration devices. 

A group of environmentalists will test to see if this is true at the 1% level of significance. 

Assume that their sample of size 50 gives a mean of 32.5 ppm.

Perform a hypothesis test at the 1% level of significance and state your decision.

### Hypothesis 
H0: μ $\geq$ 34

H1: μ $<$ 34

### Significance level
Significance Level = 1%

### Test Statics
$\mu = 34$

$\sigma = 8$

n = 50

$\bar {x}$ = 32.50

In [3]:
z = (32.5 - 34)/ (8 / (50 ** 0.5))

print(f"P(Z < {z}) = {stats.norm.cdf(z)}")

P(Z < -1.3258252147247767) = 0.0924487994828001


This is a lower tailed z test. Since 0.0924 is greater than 0.01, **we do not reject the Null Hypothesis.**

***
***

# Problem Statement 4
Based on population figures and other general information on the U.S. population, suppose it has been estimated that, on average, a family of four in the U.S. spends about $1135 annually on dental expenditures. 

Suppose further that a regional dental association wants to test to determine if this figure is accurate for their area of country. 

To test this, 22 families of 4 are randomly selected from the population in that area of the country and a log is kept of the family’s dental expenditure for one year. 

The resulting data are given below. Assuming, that dental expenditure is normally distributed in the population, use the data and an alpha of 0.5 to test the dental association’s hypothesis.

1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994

### Hypothesis 
H0: μ = \\$1135

H1: μ $\neq$ \\$1135

### Significance level
Significance Level = 5%

### Test Statics
$\mu = 1135$

$\sigma$ = Will use the sample variance as population variance

n = 22

$\bar {x}$ = mean of (1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994)

In [4]:
data = np.array([1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994])

population_mean = 1135
mean = np.mean(data)
std_deviation = np.std(data, ddof=1)
sample_size = 22
df = sample_size - 1

In [5]:
# We can assume this to be a large normally distributed population
# Since the number of samples is 22 (which is less than 30) will use the t score instead of z score

t_score = (mean - population_mean) * (sample_size ** 0.5) / std_deviation

print(f"t_score : {t_score}")

t_score : -2.023137479931484


The Critical value of t is 2.080 (from t table, t$_{.975}$ and degree of freedom = 21)

So we can accept the null hypothesis if -2.080 $\geq$ t $\leq$ 2.080

Since t score is -2.023, it lies inside the critical region, so we **accept the Null Hypothesis**

In [6]:
p = stats.t.cdf(t_score, df=df)

print(f"p = {p}")

# Since this is a two tailed test, the probability is

print(f"2 * p = {2 * p}")

p = 0.02798691597322925
2 * p = 0.0559738319464585


So the probability is 0.0559, which is greater than 0.05, so **we accept the Null Hypothesis**

***
***

# Problem Statement 5
In a report prepared by the Economic Research Department of a major bank the Department manager maintains that the average annual family income on Metropolis is $48,432. 

What do you conclude about the validity of the report if a random sample of 400 families shows and average income of $48,574 with a standard deviation of 2000?

### Hypothesis 
H0: μ = \\$48432

H1: μ $\neq$ \\$48432

### Significance level
Significance Level = 5%

### Test Statics

$\mu$ = 48432

S = 2000

n = 400

$\bar {x}$ = 48574

In [7]:
mu = 48432
s = 2000
n = 400
x = 48574

z = (x - mu) * (n ** 0.5) / s

print(f"z score = {z}")

z score = 1.42


We are using a significance level of 5%, since it is a two tailed test,

-1.96 <= z <= +1.96 so that we can reject the Alternate Hypothesis

Since z = 1.42, **we cannot reject the Null Hypothesis**

***
***

# Problem Statement 6
Suppose that in past years the average price per square foot for warehouses in the United States has been \\$32.28. 

A national real estate investor wants to determine whether that figure has changed now. The investor hires a researcher who randomly samples 19 warehouses that are for sale across the United States and finds that the mean price per square foot is \\$31.67, with a standard deviation of \\$1.29. 

Assume that the prices of warehouse footage are normally distributed in population. 

If the researcher uses a 5% level of significance, what statistical conclusion can be reached? What are the hypotheses?

In [8]:
### Hypothesis 
H0: μ = \\$32.28

H1: μ $\neq$ \\$32.28

### Significance level
Significance Level = 5%

### Test Statics

$\mu$ = 32.28

S = 1.29

n = 19

$\bar {x}$ = 31.67

SyntaxError: unexpected character after line continuation character (<ipython-input-8-84b35e3d8351>, line 2)

In [ ]:
population_mean = 32.28
x_bar = 31.67
std_deviation = 1.29
sample_size = 19
df = sample_size - 1

In [ ]:
# We can assume this to be a large normally distributed population
# Since the number of samples is 19 (which is less than 30) will use the t score instead of z score

t_score = (x_bar - population_mean) * (sample_size ** 0.5) / std_deviation

print(f"t_score : {t_score}")

The Critical value of t is 2.101 (from t table, t$_{.975}$ and degree of freedom = 18)

So we can accept the null hypothesis if -2.101 $\geq$ t $\leq$ 2.101

Since t score is -2.06, it lies inside the critical region, so we **accept the Null Hypothesis**

In [ ]:
p = stats.t.cdf(t_score, df=df)

print(f"p = {p}")

# Since this is a two tailed test, the probability is

print(f"2 * p = {2 * p}")

So the probability is 0.0540, which is greater than 0.05, so we **accept the Null Hypothesis**

***
***

# Problem Statement 7

Fill in the blank spaces in the table and draw your conclusions from it.

| Acceptance Region | Sample Size  |  $ \alpha $ | $ \beta $  at  $ \mu $ = 52   | $ \beta $  at  $ \mu $ = 50.5 |
|:--|:-:|:-:|:-:|:-:|
|  48.5 < $ \bar{x} $ < 51.5   |  10 |   |   |   |
|  48 < $ \bar{x} $ < 52       | 10  |   |   |   |
|  48.81 < $ \bar{x} $ < 51.9  |  16 |   |   |   |
|  48.42 < $ \bar{x} $ < 51.58 |  16 |   |   |   |



In [28]:
## Assuming standard_deviation is 2.5 (from http://mx.nthu.edu.tw/~cghsu/courses/ps/note2015/11_HypoTestOneSample_Note.pdf)

std_deviation = 2.5

In [30]:
## alpha for acceptance region 48.5 < x < 51.5 with sample size 10

mean = 50
n = 10
se = std_deviation / (n ** 0.5)

# alpha = probability(x_bar < 48.5) + probability(x_bar > 51.5)

z_lower = (48.5 - mean) / se
z_upper = (51.5 - mean) / se

alpha = stats.norm.cdf(z_lower) + 1 - stats.norm.cdf(z_upper)
alpha

0.05777957112359722

In [33]:
## beta at mean = 52 for acceptance region 48.5 < x < 51.5 with sample size 10

mean = 52
n = 10
se = std_deviation / (n ** 0.5)

# alpha = probability(x_bar < 48.5) + probability(x_bar > 51.5)

z_lower = (48.5 - mean) / se
z_upper = (51.5 - mean) / se

beta = stats.norm.cdf(z_upper) - stats.norm.cdf(z_lower)
beta

0.26353985497284643

In [41]:
## beta at mean = 50.5 for acceptance region 48.5 < x < 51.5 with sample size 10

mean = 50.5
n = 10
se = std_deviation / (n ** 0.5)

# alpha = probability(x_bar < 48.5) + probability(x_bar > 51.5)

z_lower = (48.5 - mean) / se
z_upper = (51.5 - mean) / se

beta = stats.norm.cdf(z_upper) - stats.norm.cdf(z_lower)
beta

0.8913423764409649


### Final Solution with the table cells filled

| Acceptance Region | Sample Size  |  $ \alpha $ | $ \beta $  at  $ \mu $ = 52   | $ \beta $  at  $ \mu $ = 50.5 |
|:--|:-:|:-:|:-:|:-:|
|  48.5 < $ \bar{x} $ < 51.5   |  10 | 0.0578  | 0.2643  | 0.8913  |
|  48 < $ \bar{x} $ < 52       | 10  |   |   |   |
|  48.81 < $ \bar{x} $ < 51.9  |  16 |   |   |   |
|  48.42 < $ \bar{x} $ < 51.58 |  16 |   |   |   |

***
***

# Problem Statement 8
Find the t-score for a sample size of 16 taken from a population with mean 10 when the sample mean is 12 and the sample standard deviation is 1.5.

In [ ]:
mean = 10
sample_mean = 12
sample_std_deviation = 1.5
sample_size = 16

t_score = (sample_mean - mean) * (sample_size ** 0.5) / sample_std_deviation

print(f"The t score is {t_score}")

***
***

# Problem Statement 9
Find the t-score below which we can expect 99% of sample means will fall if samples of size 16 are taken from a normally distributed population.

### From t table

From the t table, look at value of intersection of row df=15 and column $t_{.99}$ 

**The tscore below which we can expect 99% of sample means = 2.602**

In [ ]:
# Using scipy

print(f"The tscore below which we can expect 99% of sample means = {stats.t.ppf(.99, 15)}")

***
***

# Problem Statement 10
If a random sample of size 25 drawn from a normal population gives a mean of 60 and a standard deviation of 4, find the range of t-scores where we can expect to find the middle 95% of all sample means. Compute the probability that ($-t_{0.05} < t < t_{0.10}$).

In [ ]:
mean = 60
std_deviation = 4
sample_size = 25

print("95% of all sample means will lie between "\
      f"t = {stats.t.ppf(.025, sample_size - 1)} and t = {stats.t.ppf(.975, sample_size - 1)}")

***
***
**Compute the probability that ( $-t_{0.05} < t < t_{0.10}$ ).**

$-t_{0.05}$, will take .05 of the area to the left
$t_{0.10}$, will take .1 of the area to the right

**So the probability that ( $-t_{0.05} < t < t_{0.10}$ ) = 1 - 0.05 - 0.1 = 0.85**

***
***

# Problem Statement 11
Two-tailed test for difference between two population means

Is there evidence to conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week, given the following:

**Population 1: Bangalore to Chennai**

n1 = 1200

x1 = 452

s1 = 212

**Population 2: Bangalore to Hosur**

n2 = 800

x2 = 523

s2 = 185

***
### Solution

We will use the z test as n1 (1200) and n2 (800) are greater than 30.

$H_{0} : \mu_{1} = \mu_{2} $, ie. The number of people travelling from Bangalore to Chennai is **NOT different** from the number of people travelling from Bangalore to Hosur

$H_{1} : \mu_{1} \neq \mu_{2} $, ie.The number of people travelling from Bangalore to Chennai is **different** from the number of people travelling from Bangalore to Hosur

It will be a two tailed test

In [17]:
n1 = 1200
x1 = 452
s1 = 212

n2 = 800
x2 = 523
s2 = 185

standard_error = (((s1 ** 2) / n1) + ((s2 ** 2) / n2)) ** 0.5

z_score = (x1 - x2) / standard_error

print(f"z_score = {z_score}")

z_score = -7.926428526759299


If we take a significance level of 0.05, we will accept the Null Hypothesis, if

0.025 <= P(z_score) <= 0.975

Or

-1.96 <= z_score <= 1.96

Since z_score = -7.926, which is less than -1.96, we will **Reject the Null Hypothesis**

***
***

# Problem Statement 12

Is there evidence to conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery, given the following:

**Population 1: Duracell**

n1 = 100

x1 = 308

s1 = 84

**Population 2: Energizer**

n2 = 100

x2 = 254

s2 = 67

***
### Solution

We will use the z test as n1 (100) and n2 (100) are greater than 30.

$H_{0} : \mu_{1} = \mu_{2} $, ie. People do not prefer Duracell over Energizer Battery

$H_{1} : \mu_{1} \neq \mu_{2} $, ie. People do prefer Duracell over Energizer Battery

It will be a two tailed test

In [18]:
n1 = 100
x1 = 308
s1 = 84

n2 = 100
x2 = 254
s2 = 67

standard_error = (((s1 ** 2) / n1) + ((s2 ** 2) / n2)) ** 0.5

z_score = (x1 - x2) / standard_error

print(f"z_score = {z_score}")

z_score = 5.025702668336442


If we take a significance level of 0.05, we will accept the Null Hypothesis, if

0.025 <= P(z_score) <= 0.975

Or

-1.96 <= z_score <= 1.96

Since z_score = 5.0257, which is greater than 1.96, we will **Reject the Null Hypothesis**

***
***

# Problem Statement 13
Pooled estimate of the population variance

Does the data provide sufficient evidence to conclude that average percentage increase in the price of sugar differs when it is sold at two different prices?


**Population 1: Price of sugar = Rs. 27.50**

n1 = 14

x1 = 0.317%

s1 = 0.12%

**Population 2: Price of sugar = Rs. 20.00**

n2 = 9

x2 = 0.21%

s2 = 0.11%

***
### Solution

We will use the t test as n1 (14) and n2 (9) are less than 30.

$H_{0} : \mu_{1} = \mu_{2} $

$H_{1} : \mu_{1} \neq \mu_{2} $

It will be a two tailed test

In [24]:
n1 = 14
x1 = 0.317
s1 = 0.12

n2 = 9
x2 = 0.21
s2 = 0.11

pooled_variance = (((n1 - 1) * (s1 ** 2)) + ((n2 - 1) * (s2 ** 2))) / (n1 + n2 - 2)

n12 = (1 / n1) + (1 / n2)

t_score = (x1 - x2) / ((pooled_variance * n12) ** 0.5)

t_score

2.15355322387416

If we take a 0.05 significance level, looking at t table, for df = 21, t = 2.080

Since the calculated t is 2.1534, more than the critical t (= 2.080), **we reject the Null Hypothesis**

***
***

# Problem Statement 14
The manufacturers of compact disk players want to test whether a small price reduction is enough to increase sales of their product. 

Is there evidence that the small price reduction is enough to increase sales of compact disk players?


**Population 1: Before reduction**

n1 = 15

x1 = Rs. 6598

s1 = Rs. 844

**Population 2: After reduction**

n2 = 12

x2 = RS. 6870

s2 = Rs. 669

***
### Solution

We will use the t test as n1 (15) and n2 (12) are less than 30.

$H_{0} : \mu_{1} \geq \mu_{2} $

$H_{1} : \mu_{1} < \mu_{2} $

It will be a upper tailed test

In [26]:
n1 = 15
x1 = 6598
s1 = 844

n2 = 12
x2 = 6870
s2 = 669

pooled_variance = (((n1 - 1) * (s1 ** 2)) + ((n2 - 1) * (s2 ** 2))) / (n1 + n2 - 2)

n12 = (1 / n1) + (1 / n2)

t_score = (x2 - x1) / ((pooled_variance * n12) ** 0.5)

t_score

0.9098300343990461

If we take a 0.05 significance level, looking at t table, for df = 25, t = 1.708

Since the calculated t is 0.909, less than the critical t (= 1.708), **we cannot reject the Null Hypothesis**

***
***

# Problem Statement 15
Comparisons of two population proportions when the hypothesized difference is zero

Carry out a two-tailed test of the equality of banks’ share of the car loan market in 1980 and 1995.

**Population 1: 1980**

n1 = 1000

x1 = 53

$\hat{p}$1 = 0.53

**Population 2: 1985**

n2 = 100

x2 = 43

$\hat{p}$2= 0.53

***
***

# Problem Statement 16
Carry out a one-tailed test to determine whether the population proportion of traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes are offered as at least 10% higher than the proportion of such buyers when no sweepstakes are on.


**Population 1: With sweepstakes**

n1 = 300

x1 = 120

$\hat{p}$1 = 0.40

**Population 2: No sweepstakes**

n2 = 700

x2 = 140

$\hat{p}$2= 0.20

***
***

# Problem Statement 17
A die is thrown 132 times with the following results:

|   |   |   |   |   |  |   |
|:--|:--|:--|:--|:--|:--|:--|
|Number turned up: | 1| 2| 3| 4| 5| 6|
|Frequency: | 16| 20| 25| 14| 29| 28|

Is the die unbiased? Consider the degrees of freedom as $\hat{p} - 1$.

***
***

# Problem Statement 18
In a certain town, there are about one million eligible voters. A simple random sample of 10,000 eligible voters was chosen to study the relationship between gender and participation in the last election. 

The results are summarized in the following 2X2 (read two by two) contingency table:

| | Men | Women |
|--|:--:|:--:|
|voted|2792|3591|
|not voted|1486|2131|


We would want to check whether being a man or a woman (columns) is independent of having voted in the last election (rows). 

In other words, is “gender and voting independent”?

***
***

# Problem Statement 19
A sample of 100 voters are asked which of four candidates they would vote for in an election. The number supporting each candidate is given below:

|Higgins|Reardon|White|Charlton|
|:-:|:-:|:-:|:-:|
|41|19|24|16|

Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96, with 3 df, $\hat{p}$ < 0.05].

***
***

# Problem Statement 20
Children of three ages are asked to indicate their preference for three photographs of adults. Do the data suggest that there is a significant relationship between age and photograph preference? 

<table>
    <thead>
        <tr>
            <th rowspan=2 colspan=2></th>
            <th colspan=3>Photograph</th>
        </tr>
        <tr>
            <th>A</th>
            <th>B</th>
            <th>C</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td rowspan=3>Age of Child</td>
            <td>5-6 years</td> 
            <td>18</td> 
            <td>22</td> 
            <td>20</td>
        </tr>
        <tr>
            <td>7-8 years</td> 
            <td>2</td> 
            <td>28</td> 
            <td>40</td>
        </tr>
        <tr>
            <td>9-10 years</td> 
            <td>20</td> 
            <td>10</td> 
            <td>40</td>
        </tr>
    </tbody>
</table>

What is wrong with this study? [Chi-Square = 29.6, with 4 df: $\hat{p}$ < 0.05].


***
***

# Problem Statement 21
A study of conformity using the Asch paradigm involved two conditions: one where one confederate supported the true judgement and another where no confederate gave the correct response.

| | Support | No Support|
|:--|:--:|:--:|
|Conform|18|40|
|Not conform|32|10|

Is there a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df: $\hat{p}$  < 0.05].

***
***

# Problem Statement 22

We want to test whether short people differ with respect to their leadership qualities (Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget MP's are there?) The following table shows the frequencies with which 43 short people and 52 tall people were categorized as "leaders", "followers" or as "unclassifiable". 

Is there a relationship between height and leadership qualities?


[Chi-Square = 10.71, with 2 df: $\hat{p}$ < 0.01].

<table>
    <thead>
        <tr>
            <th rowspan=2 colspan=></th>
            <th colspan=2>Height</th>
        </tr>
        <tr>
            <th>Short</th>
            <th>Tall</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Leader</td> 
            <td>12</td> 
            <td>32</td>
        </tr>
        <tr>
            <td>Follower</td> 
            <td>22</td> 
            <td>14</td> 
        </tr>
        <tr>
            <td>Unclassifiable</td> 
            <td>9</td> 
            <td>6</td>
        </tr>
    </tbody>
</table>

***
***

# Problem Statement 23

Each respondent in the Current Population Survey of March 1993 was classified as employed, unemployed, or outside the labor force. The results for men in California age 35-44 can be cross-tabulated by marital status, as follows:

|  | Married | Widowed, Divorced or Separated | Never Married|
|:-|:--|:--|:--|
|Employed|679|103|114|
|Unemployed|63|10|20|
|Not in labor force|42|18|25|

Men of different marital status seem to have different distributions of labor force status. Or is this just chance variation? (you may assume the table results from a simple random sample.)